src/types.ts

```diff
@@ -14,16 +15,29 @@ export const enum TagAttributes {
   LOCALE = 'data-contentful-locale',
 }
 
-// TODO: this has kind of overlap with CollectionItem, can we combine them?
-export type Entity = Record<string, unknown>;
-export type Argument = Entity | Entity[];
-export type SubscribeCallback = (data: Argument) => void;
-
 export interface SysProps {
   id: string;
   [key: string]: unknown;
 }
 
+export type Entity = Record<string, unknown>;
+export interface EntityWithSys extends Entity {
+  sys: SysProps;
+  __typename?: string;
+}
+
+export function hasSysInformation(entity: unknown): entity is EntityWithSys {
+  return !!(
```

Comment:
`entity && typeof entity === 'object' && entity.satisfies((e) => 'sys' in e && e.sys.id);`

Just showing, no need to change

Reply:
I wonder would `!!entity?.satisfies((e) => 'sys' in e && e.sys.id)` even work?